raster-foundry/raster-foundry
# Enable admins posting org logo and storing it on S3

## Overview

This PR:
 * creates a migration to add a column in organizations table to store S3 uri of org logo
 * enables posting to `organizations/<orgid>/logo` endpoint to upload logo to S3 and store the uri in the org table
 * makes the logo: 
     - public-read
     - overwrite old logos of the same org
     - keyed by org id
 * allows only org admins to do so

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Notes

Per offline discussion, PNG and resolution restrictions seem better to fall on frontend. But open to further suggestions.


## Testing Instructions

 * Run migration.
 * Log in as dev user and grab a token.
 * Use whatever you want to convert a small png to base64 string
 * `POST` to `api/organizations/<organization ID>/logo` with body like
```json
{
     "logo": "<your png as base 64 string>"
}
```
 * You access should be denied  since dev user is `MEMBER` of the public organization
 * Open a sql shell and:
```sql
update user_group_roles set group_role = 'ADMIN' where user_id = 'auth0|59318a9d2fbbca3e16bcfc92';
```
 * Do the `POST` again, and you should see the updated organization with logo S3 uri. Check it out in the browser to make sure it is public.
  * Open a sql shell and:
```sql
update user_group_roles set group_role = 'MEMBER' where user_id = 'auth0|59318a9d2fbbca3e16bcfc92';
```

Closes #3385
